config-model/src/test/java/com/yahoo/vespa/model/application/validation/change/search/AttributeChangeValidatorTest.java

```diff
@@ -190,27 +201,83 @@ void changing_distance_metric_with_hnsw_index_enabled_requires_restart() throws
-                        "'max-links-per-node' from '16' to '4'"));
+    void changing_hnsw_index_property_max_links_per_node_requires_restart() {
+        var currentSd = """
+                field f1 type tensor(x[2]) {
+                  indexing: attribute | index
+                  index {
+                    hnsw {
+                    }
+                  }
+                }
+                """;
+        var nextSd = """
+                field f1 type tensor(x[2]) {
+                  indexing: attribute | index
+                  index {
+                    hnsw {
+                      max-links-per-node: 4
+                    }
+                  }
+                }
+                """;
+
+        var expectedOutput = "Field 'f1' changed: change hnsw index property 'max-links-per-node' from '16' to '4'";
+
+        // Success when validation override is set
+        new Fixture(currentSd, nextSd, ValidationId.hnswSettingsChange)
+                .assertValidation(newRestartAction(ClusterSpec.Id.from("test"), expectedOutput));
+
+        // Should fail when validation override is not set
+        try {
+            new Fixture(currentSd, nextSd).assertValidation(newRestartAction(ClusterSpec.Id.from("test"), expectedOutput));
+            fail("Expected exception on changing hnsw settings");
+        }
+        catch (ValidationOverrides.ValidationException e) {
+            assertTrue(e.getMessage().contains(ValidationId.hnswSettingsChange.toString()));
+        }
     }
 
     @Test
-    void changing_hnsw_index_property_neighbors_to_explore_at_insert_requires_restart() throws Exception {
-        new Fixture("field f1 type tensor(x[2]) { indexing: attribute | index \n index { hnsw } }",
-                "field f1 type tensor(x[2]) { indexing: attribute | index \n index { " +
-                        "hnsw { neighbors-to-explore-at-insert: 100 } } }").
-                assertValidation(newRestartAction(ClusterSpec.Id.from("test"),
-                "Field 'f1' changed: change hnsw index property " +
-                        "'neighbors-to-explore-at-insert' from '200' to '100'"));
+    void changing_hnsw_index_property_neighbors_to_explore_at_insert_requires_restart() {
+        var currentSd = """
+                field f1 type tensor(x[2]) {
+                  indexing: attribute | index
+                  index {
+                    hnsw {
+                    }
+                  }
+                }
+                """;
+        var nextSd = """
+                field f1 type tensor(x[2]) {
+                  indexing: attribute | index
+                  index {
+                    hnsw {
+                      neighbors-to-explore-at-insert: 100
+                    }
+                  }
+                }
+                """;
+
+        var expectedOutput = "Field 'f1' changed: change hnsw index property 'neighbors-to-explore-at-insert' from '200' to '100'";
```

Comment:
> Adding or removing the [hnsw index](https://docs.vespa.ai/en/reference/schema-reference.html?mode=cloud#index-hnsw) on a tensor attribute field, or changing the **distance-metric** or max-links-per-node requires a restart to rebuild the index. **Changing neighbors-to-explore-at-insert requires a restart, but does not rebuild the index.**

Suggest we only require validation for those things that rebuild the index (make sure to include change of distance-metric)

Reply:
Yes, that makes more sense, thanks